packages/experimental/ActivityIndicator/src/ActivityIndicator.tsx

```diff
@@ -0,0 +1,95 @@
+import { activityIndicatorName, ActivityIndicatorProps, ActivityIndicatorType } from './ActivityIndicator.types';
+import { diameterSizeMap, lineThicknessSizeMap, stylingSettings } from './ActivityIndicator.styling';
+
+// Can also just put this within the compose, would that be better?
+const getActivityIndicatorPath = (diameter: number, width: number, color: string) => {
+  const start = {
+    x: width / 2,
+    y: diameter / 2,
+  };
+  const innerRadius = diameter / 2 - width / 2;
+  const path = `M${start.x} ${start.y} a${innerRadius} ${innerRadius} 0 1 1 ${innerRadius} ${innerRadius}`;
+
+  return <Path d={path} stroke={color} strokeWidth={width} strokeLinecap="round" />;
+};
+
+export const AnimatedSvg = Animated.createAnimatedComponent(Svg);
+const useStyling = buildUseStyling(stylingSettings);
+export const ActivityIndicator = compose<ActivityIndicatorType>({
+  displayName: activityIndicatorName,
+  ...stylingSettings,
+  slots: {
+    root: AnimatedSvg,
+  },
+  render: (props: ActivityIndicatorProps, useSlots: UseSlots<ActivityIndicatorType>) => {
+    const Slots = useSlots(props);
+    const tokens = useStyling(props).root;
+
+    const memoizedActivityIndicatorData = useMemo(
+      () => ({
+        activityIndicatorColor: props.activityIndicatorColor ? props.activityIndicatorColor : tokens['activityIndicatorColor'],
+        size: props.size ? diameterSizeMap[props.size] : diameterSizeMap[tokens['size']],
+        lineThickness: props.lineThickness
+          ? lineThicknessSizeMap[props.lineThickness]
+          : props.size
+          ? lineThicknessSizeMap[props.size]
+          : lineThicknessSizeMap[tokens['lineThickness']],
+        animating: props.animating != undefined ? props.animating : true,
+        hidesWhenStopped: props.hidesWhenStopped != undefined ? props.hidesWhenStopped : true,
+      }),
+      [],
+    );
+
+    const spinAnimation = useRef(new Animated.Value(0)).current;
+    useEffect(() => {
+      if (memoizedActivityIndicatorData.animating) {
+        Animated.loop(
+          Animated.timing(spinAnimation, {
+            toValue: 359,
+            duration: 750,
+            useNativeDriver: false,
+            easing: Easing.linear,
+          }),
+        ).start();
+      }
+    });
+    const interpolateSpin = spinAnimation.interpolate({
+      inputRange: [0, 359],
+      outputRange: ['0deg', '359deg'],
+    });
+
+    // React Native ActivityIndicator still takes up space when hidden, so to perfectly match would use opacity
+    // hiding opacity makes the screen reader on iOS and Android skip over it
+    const hideOpacity = memoizedActivityIndicatorData.animating == false && memoizedActivityIndicatorData.hidesWhenStopped == true ? 0 : 1;
+
+    const path = getActivityIndicatorPath(
+      memoizedActivityIndicatorData.size,
+      memoizedActivityIndicatorData.lineThickness,
+      memoizedActivityIndicatorData.activityIndicatorColor as string,
+    );
```

Comment:
I think this is actually something that would be better benefit from useMemo, since we'll probably generate the same path multiple times.

Reply:
I'm not sure I can put this within memo because it depends on size, line thickness, and color resolving themselves in memo first. If it's within memo, they will also need to resolve themselves as arguments to the getPath function.

Reply:
Yeah that makes sense. IMO, it feels like the call to useMemo is getting in the way of writing a cleaner / more performant implementation, because we're caching the easy to compute stuff but not the hard-to-compute stuff (path). I would personally look to see if we can just remove it. When I used it in Avatar, there were different requirements around that control that encouraged us adding that call. I don't think it makes as much sense for ActivityIndicator anymore.